pkg/controller/controller.go

```diff
@@ -169,6 +169,21 @@ func (w *WorkflowController) enqueue(workflow *wapi.Workflow) {
 	w.queue.Add(key)
 }
 
+// get workflow by key method
+func (w *WorkflowController) getWorkflowByKey(key string) (*wapi.Workflow, error) {
```

Comment:
You should be able to use this method also in the beginning of the `(w *WorkflowController) sync(key string) error` method

Reply:
Agree. Will update.

Reply:
Done.